### ISSUE TITLE
Does not work with GHC 9.10.1 / Linux

### ISSUE THREAD

**Issue by sol** [haskell-fswatch/hfsnotify#115]
Steps to reproduce:

```
$ cabal test
```

All tests for the native implementation fail with:

```
Error: couldn't start native file manager: fdType: unsupported operation (unknown file type)
```

**Comment by sol** [comment]
This has been reported here: https://github.com/kolmodin/hinotify/issues/38

But the underlying reason is: https://github.com/ghc/ghc/commit/e206be6470d6c1d4a5ed80615c8dc830efe011ea#diff-97f1e321bbe75ce776f98261d832c63215aa181752861f8523988a4bc23187caR473

Which ultimately runs into the other otherwise case here:

https://github.com/ghc/ghc/blob/7e02111b71609bd4859d09c23482ea947fc35729/libraries/ghc-internal/src/GHC/Internal/System/Posix/Internals.hs#L153

**Comment by danwdart** [comment]
ghcid uses this - hence I figure that this is important. For now I'm downgrading ghcid's compiler.

**Comment by thomasjm** [comment]
Yep, we're waiting to pick up https://github.com/kolmodin/hinotify/pull/39 from upstream.

**Comment by thomasjm** [comment]
This was fixed in 011d8234b78bbfa5d3717fb3af435c4ba21f97fd and will be released in `0.4.2.0`. (Thanks for your efforts in fixing this in `hinotify`!)
